main_2024.py

```diff
@@ -255,6 +276,130 @@ def main() -> int:
                 print("geolocation confidence: " + str(detection_world.confidence))
                 print("")
 
+        # Check if all workers are alive in each manager.
+        # If a worker is dead, log the error and restart the worker.
+        # Steps: Terminate and join the worker, fill and drain the
+        # related queues, and create and start a new worker.
+        for manager in range(number_of_managers):
```

Comment:
you should create a helper function for this - on an error, pass in the needed info to restart the worker

Reply:
Would you be able to clarify what I should create the helper function for? Would you like me to create a helper function for 1 worker, for each worker, or for the for loop?

Reply:
there are some common elements to restarting a dead worker

1. we log a message
2. we drain its input queue(s) (since the workers have difference numbers of queues, pass an array of queues to drain)
3. delete the manager
4. create a new manager and add it to the array of managers
5. create workers for the new manager

I was thinking depending on the ManagerType, we would pass different parameters into a function that does this, maybe smth like terminate_and_restart_worker()

You could also have another helper function that does the for loop with the if blocks and calls terminate_and_restart_worker, but that seems unnecessary and you would have to pass it basically everything which is unideal